tests/prague/eip3074_auth_authcall/test_auth.py

```diff
@@ -0,0 +1,496 @@
+"""
+Tests for EIP-3074's `AUTH` Instruction
+"""
+
+from typing import Tuple
+
+import pytest
+from coincurve.keys import PrivateKey, PublicKey
+from ethereum.base_types import U256
+from ethereum.crypto.hash import keccak256
+
+from ethereum_test_tools import (
+    Account,
+    Address,
+    Environment,
+    StateTestFiller,
+    TestAddress,
+    TestAddress2,
+    TestPrivateKey,
+    TestPrivateKey2,
+    Transaction,
+)
+from ethereum_test_tools.common.base_types import Hash
+from ethereum_test_tools.vm.opcode import Opcodes as Op
+
+REFERENCE_SPEC_GIT_PATH = "EIPS/eip-3074.md"
+REFERENCE_SPEC_VERSION = "3b5fcad6b35782f8aaeba7d4ac26004e8fbd720f"
+
+MAGIC = b"\x04"
+
+
+@pytest.mark.valid_from("Prague")
+def test_all_zeros(state_test: StateTestFiller) -> None:
+    """
+    Test `AUTH` opcode with all zero arguments.
+    """
+    env = Environment()
+    to = Address(0x100)
+    pre = {
+        to: Account(code=Op.SSTORE(1, Op.ISZERO(Op.AUTH(0, 0, 0)))),
+        TestAddress: Account(balance=1000000000000000000000),
+    }
+    tx = Transaction(
+        to=to,
+        gas_limit=50000,
+    )
+    post = {
+        to: Account(storage={"0x01": "0x01"}),
+    }
+
+    state_test(env=env, pre=pre, post=post, tx=tx)
+
+
+@pytest.mark.valid_from("Prague")
+def test_stack_underflow(state_test: StateTestFiller) -> None:
```

Comment:
I remember here Ori asked if it is possible to get vmtrace, because it contains an error code that we want to check here. such as stack underflow. because otherwise the test could potentially fail with OOG and we didn't check the exception